lib/services/keyVaultManagement/lib/models/index.d.ts

```diff
@@ -191,15 +335,15 @@ export interface VaultCreateOrUpdateParameters extends BaseResource {
  *
  * @member {string} [id] The Azure Resource Manager resource ID for the key
  * vault.
- * @member {string} name The name of the key vault.
+ * @member {string} [name] The name of the key vault.
  * @member {string} [type] The resource type of the key vault.
  * @member {string} location The supported Azure location where the key vault
  * should be created.
  * @member {object} [tags] The tags that will be assigned to the key vault.
  */
 export interface Resource extends BaseResource {
   readonly id?: string;
-  name: string;
+  readonly name?: string;
```

Comment:
Marking an existing interface property as readonly is a breaking change. Requires major version increment.